### PR TITLE
Memory Growth Tests for Java API

### PR DESCRIPTION
**Pull request by dyastremsky**
Add a test to validate that memory growth does not occur over time for the Java bindings.

### CODE REVIEW

**Comment by dyastremsky** [comment]
Currently, there is high memory growth (from 25MB to 1.7GB). I am investigating to see if I can find a memory leak or any issues within the test. The example is based on the simple.java example used as a sample.

**Comment by CoderHam** [comment]
Maybe I missed this but why do we elect to not put on java simple example in the client repo?

**Comment by dyastremsky** [comment]
> Maybe I missed this but why do we elect to not put on java simple example in the client repo?

This is for the Java API, not the Java client. Will defer to @jbkyang-nvi in case the client makes more sense.

**Comment by jbkyang-nvi** [comment]
We plan to move all of [javacpp-presets/tritonserver] related things to Triton eventually. I think its reasonable to have only 1 copy for now, before we move everything over.

File: qa/L0_java_memory_growth/Simple.java [hunk 1]
```diff
@@ -0,0 +1,1018 @@
...
+          if(ValidateMemoryGrowth(.1f)){
+            System.out.println("Memory growth test passed");
+          } else {
+            System.out.println("Memory growth test FAILED");
+          }
+        };
+      Thread memory_thread = new Thread(runnable);
+      memory_thread.start();
+
+      for(int i = 0; i < 1000000; i++){
```
Comment (GuanLuo) [inline qa/L0_java_memory_growth/Simple.java:1008]: Can you expose command line option to set the number of iteration? Then we can indirectly control the length of the test in `test.sh`

Comment (dyastremsky) [reply qa/L0_java_memory_growth/Simple.java]: Good recommendation. Done.

**Review by jbkyang-nvi — COMMENTED** [review]
Can we have a single instance of Simple, and call different functions for each instance? It is a bit hard to review all of them.

**Comment by jackyh** [comment]
as you know, I've already tried the "vibration version", since the mem allocated will still has some range of 0 to 100MB, so that sort of test will still fail. But, that should not be an issue, since it will not grow gradually.

**Comment by dyastremsky** [comment]
Closing due to https://github.com/triton-inference-server/server/pull/4259 (i.e. Java API memory growth test) being merged.

**Pull request closed without merging**